src/components/Canvas.vue

```diff
@@ -106,11 +111,15 @@ export default {
     },
     Object.assign.apply(null, [{}, ...Object.values(nodeTypesFlat)])
   ),
+  props: {
+    nodes: { type: Array, default: () => [] },
+    edges: { type: Array, default: () => [] }
+  },
   data () {
     return {
+      isMounted: false,
       nodeTypes,
-      nodes: [],
-      edges: [],
+      connectors: {},
```

Comment:
A cache for all the connectors on the canvas mapped from their ids (must not always include all at any time).